tests/operator_test.go

```diff
@@ -234,8 +287,107 @@ var _ = Describe("Operator delete CDI tests", func() {
 		err = f.CdiClient.CdiV1beta1().CDIs().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{DryRun: []string{"All"}})
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	It("Should install CDI infrastructure pods with node placement", func() {
+		By("Deleting CDI CR")
+		err := f.CdiClient.CdiV1beta1().CDIs().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for CDI CR and infra deployments to be deleted after CDI CR was removed")
+		Eventually(func() bool { return infraDeploymentGone(f) && crGone(f, cr) }, 15*time.Minute, 2*time.Second).Should(BeTrue())
+
+		By("Deleting CDI operator")
+		err = f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Delete(context.TODO(), "cdi-operator", metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for CDI operator to be deleted")
+		Eventually(func() bool { return cdiOperatorGone(f) }, 5*time.Minute, 2*time.Second).Should(BeTrue())
+
+		By("Creating modified CDI CR, with infra nodePlacement")
+		localSpec := cr.Spec.DeepCopy()
+		localSpec.Infra = tests.TestNodePlacementValues(f)
+
+		cdi := &cdiv1.CDI{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cdi",
+			},
+			Spec: *localSpec,
+		}
+
+		_, err = f.CdiClient.CdiV1beta1().CDIs().Create(context.TODO(), cdi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Recreating CDI operator")
```

Comment:
None of this gets executed if there was a failure above for whatever reason, and now the CDI CR is gone, and all the other tests will fail. Can you put the delete and create CR in a Before and After each so it gets executed even if the test fails?

Reply:
I changed it to be a lot cleaner and with its own after/before test.